app/src/components/misc/JoinWorkspaceCard/index.tsx

```diff
@@ -0,0 +1,146 @@
+import { useState, useMemo, useEffect } from "react";
+import { useSelector, useDispatch } from "react-redux";
+import { getUserAuthDetails } from "store/selectors";
+import { Row, Avatar, Typography } from "antd";
+import { RQButton } from "lib/design-system/components";
+import { CloseOutlined } from "@ant-design/icons";
+import { httpsCallable, getFunctions } from "firebase/functions";
+import { getPendingInvites } from "backend/workspace";
+import { getDomainFromEmail, isCompanyEmail } from "utils/FormattingHelper";
+import { isEmailVerified } from "utils/AuthUtils";
+import { actions } from "store";
+import { capitalize } from "lodash";
+import { Invite, User } from "types";
+import "./index.css";
+
+export const JoinWorkspaceCard = () => {
+  const dispatch = useDispatch();
+  const user = useSelector(getUserAuthDetails);
+  const getOrganizationUsers = useMemo(() => httpsCallable(getFunctions(), "users-getOrganizationUsers"), []);
+  const userEmailDomain = useMemo(() => getDomainFromEmail(user?.details?.profile?.email)?.split(".")[0], [
+    user?.details?.profile?.email,
+  ]);
+
+  const [isBusinessDomainUser, setIsBusinessDomainUser] = useState(false);
+  const [organizationMembers, setOrganizationMembers] = useState(null);
+  const [hasActiveWorkspace, setHasActiveWorkspace] = useState(false);
+  const [hasEmailInvite, setHasEmailInvite] = useState(false);
+
+  const handleNudgeCTAClick = () => {
+    if (hasActiveWorkspace) {
+      dispatch(
+        actions.toggleActiveModal({
+          modalName: "joinWorkspaceModal",
+          newValue: true,
+          newProps: {
+            callback: () => dispatch(actions.updateJoinWorkspaceCardVisible(false)),
+          },
+        })
+      );
+    } else {
+      dispatch(
+        actions.toggleActiveModal({
+          modalName: "createWorkspaceModal",
+          newValue: true,
+          newProps: {
+            callback: () => {
+              dispatch(actions.updateJoinWorkspaceCardVisible(false));
+            },
+          },
+        })
+      );
+    }
+  };
```

Comment:
let's break this into two functions:
1. showJoinWorkspaceModal
2. showCreateWorkspaceModal